src/GNDStk/BodyText/test/types.test.cpp

```diff
@@ -0,0 +1,121 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+
+using namespace njoy::GNDStk::core;
+
+// -----------------------------------------------------------------------------
+// Scenario
+// -----------------------------------------------------------------------------
+
+SCENARIO("BodyText data types") {
+   GIVEN("std::vector<T> for certain specific Ts") {
+
+      WHEN ("Via BodyText, we make variants of vectors and of scalars") {
+         // vector type
+         using vv = BodyText<true>::VariantOfVectors;
+         THEN ("The variant-of-vector size should be correct") {
+            CHECK(std::variant_size_v<vv> == 15);
+         }
+         vv vectors;
+
+         // scalar type
+         using vs = BodyText<true>::VariantOfScalars;
+         THEN ("The variant-of-scalar size should be correct") {
+            CHECK(std::variant_size_v<vv> == 15);
+         }
+         vs scalars;
+
+         THEN ("Transformation to vector<string>, and access, work") {
+            vectors = std::vector<std::string>(10,"a");
+            scalars = std::get<std::vector<std::string>>(vectors)[0];
+            CHECK(std::get<std::string>(scalars) == "a");
```

Comment:
Perhaps I should remove this test file altogether. The idea of having it was that BodyText provides several possibilities for the vector of body text values - and users will begin using the vector types they want. If a future maintainer of GNDStk goes into the BodyText class and decides to change what vector alternatives it provides, they might inadvertently remove something that somebody, somewhere, is using. That of course wouldn't be good. By having this test, we might slow down a maintainer who tries to change what BodyText currently provides, so that they don't inadvertently remove something that someone might be using. I suppose that this could be done in a shorter manner in this test file.